Add copyright checker

BSchilperoort opened the pull request:
Adds https://github.com/eWaterCycle/copyright-checker to the CI of this repo.

BSchilperoort committed:
Make book building depend on passing copyright check

BSchilperoort committed:
Check copyright on PR level

BSchilperoort committed:
Remove separate PR-level check

BSchilperoort committed:
Revert "Remove separate PR-level check"

This reverts commit c72260c7a82eae815347d6a266a9e59db95089d0.

BSchilperoort commented:
@RolfHut @MarkMelotto this PR adds the copyright checker. In its current behavior it expects a yml file with _every_ image. 

Obviously this fails now.

I have set up branch protection rules for `main` now. This should prevent any direct pushes to main, or merges from PRs that don't have approval from a reviewer. You could consider also selecting the rule "Require status checks to pass" to enforce this checker.

MarkMelotto commented:
@BSchilperoort I am adding 2 thesis projects first and the tutorial stuff. Then I can take a look at the checker and add the correct files, hopefully tomorrow afternoon

RolfHut commented:
I will not have time to look at this in detail till after the easter weekend, sorry

MarkMelotto committed:
updating copyright checker branch (#21)

* first fix for tutorials, cleaning up the HBV code

* more fixes and linter stuff

* added both pdf thesises

* samenvatting toegevoegd

* added Gelt thesis

* added Gelt links and the updated ToC

* updated eline up until methodology

* added server launch button to projects for EGU

* added the tutorial to the intro.md

* added tutorial to the readme

* fixed figure

* added correct folder

* small change

* small changes, fixed 1 tutorial, should work now out of the box

* fixed pcrglob, but plots are still bad

* added full date for now and changed the printing to a statusbar

* cleaned up the pathing, which is probably still bad and not working, so let me check that tomorrow.
+ added hbv modelled discharge
+ added correct units to pcr

* update to tutorials md for EGU

* updated PCRglob notebook to work for EGU

* updated Elines thesis

* updated the figures of Ischa

* fix to ToC

* small changes and fixes for bigger more complicated tables

* table test

* table test

* table test 3

* table test 4

* table test 5

* table test 6

* table test 7

* table test 8

* table test 9

* table test 10 final

* final updates on BSc thesis projects for a while

* quick build fix

* added in the comments of Rolf

* added in the comments of Rolf, now also to PCR

* using correct forcing path now

* ran the notebooks

* updated changelog

* - updated changelog
- removed redundant files of Thirza
- Added Zoe

* reverted back to comments instead of docstring

* Fixed an error

Fixed an error when setting the PCR data points

* fixed jupyterhub launch image

* small fix to Gelt

* Gelt title is now in english
added Zoë to the list

---------

Co-authored-by: Mark Melotto <m.melotto@tudelft.nl>
Co-authored-by: Rolf Hut <r.w.hut@tudelft.nl>

MarkMelotto committed:
updated thirza figures

MarkMelotto committed:
updated Ischa figures

MarkMelotto committed:
updated Eline figures

MarkMelotto committed:
updated Gelt figures

MarkMelotto committed:
updated eWC figures

MarkMelotto committed:
updated less relevant figures

MarkMelotto committed:
removed some figures

MarkMelotto committed:
last fix (I hope)
updated changelog

MarkMelotto committed:
git is being annoying

MarkMelotto committed:
resolved conflicts

MarkMelotto committed:
changed names again

MarkMelotto committed:
fixed some small things again

MarkMelotto commented:
@BSchilperoort maybe it is nice to put the yml files in a separate folder as apposed to how it is now?
Suggestion:
-content/image.png
-content/copyright/image.png.yml
